RocketChat/Rocket.Chat.ReactNative
# [FIX] Messages showing URL preview on Room Actions

## Proposed changes

When we look for pinned, mentions and starred messages in their respective views, and the message contains URL, the the URL preview gets rendered. Also when we search for messages that have URL in them, we can see URL preview there also.
## Issue(s)	

Closes #4064 
Closes #4061
## How to test or reproduce

1. Go to a chat and pin and star a message containing a URL.
2. Mention yourself and add a URL to that message.
3. Go to Room Actions >  (Either Pinned / Starred / Mentions screen) and check for the message containing the URL.
4. Search for that starred message in the chat.
Now after the changes made the URL preview can be seen for starred, pinned and mentions messages view.
## Screenshots
### Before
<img width="258" alt="Screenshot 2022-04-17 at 8 41 13 PM" src="https://user-images.githubusercontent.com/55805094/163848883-14d4d796-f3f8-4a12-a53f-1e11d9cb4524.png">
<img width="258" alt="Screenshot 2022-04-17 at 8 41 19 PM" src="https://user-images.githubusercontent.com/55805094/163848885-61ae9b94-ae04-45fe-af90-d9bc452b998b.png">
<img width="258" alt="Screenshot 2022-04-17 at 8 41 24 PM" src="https://user-images.githubusercontent.com/55805094/163848888-557a890f-5313-4591-8897-e3b15d07a984.png">
<img width="446" alt="163467473-0f4911ea-c0e3-44bd-a036-fbeea62bace7" src="https://user-images.githubusercontent.com/55805094/163849100-b187a5c6-f12d-49e9-8ae3-328c8236c3c2.png">


### After
<img width="258" alt="Screenshot 2022-04-17 at 11 14 58 PM" src="https://user-images.githubusercontent.com/55805094/163726220-8da91819-e977-40c6-8cbd-eba2df11f8a5.png">
<img width="258" alt="Screenshot 2022-04-17 at 11 14 49 PM" src="https://user-images.githubusercontent.com/55805094/163726221-b2166f92-d71c-439b-b13e-216898b913c1.png">
<img width="258" alt="Screenshot 2022-04-17 at 11 14 44 PM" src="https://user-images.githubusercontent.com/55805094/163726222-0868d89e-0d80-4ae8-92ff-ea2d38e8bbc2.png">
<img width="245" alt="163540006-e81f89c8-f1f9-4758-ba7a-2db2fd3e2914" src="https://user-images.githubusercontent.com/55805094/163848806-5ddcac3e-93d1-46a0-b52c-255927f3116b.png">


## Types of changes




## Checklist



## Further comments
